gpu-direct/rhel/entrypoint.sh

```diff
@@ -1,22 +1,128 @@
 #!/bin/bash -x
 MOFED=/run/mellanox/drivers
-NVIDIA=/run/nvidia/driver
+NVIDIA=/run/nvidia/drivers
 KERNEL_VERSION=$(uname -r)
-ln -sf ${MOFED}/usr/src/ofa_kernel /usr/src/ofa_kernel
-ln -sf ${NVIDIA}/usr/src/nvidia-* /usr/src/.
-mkdir -p /lib/modules/${KERNEL_VERSION}
-ln -sf /usr/src/kernels/${KERNEL_VERSION} /lib/modules/${KERNEL_VERSION}/build
-touch /lib/modules/${KERNEL_VERSION}/modules.order
-touch /lib/modules/${KERNEL_VERSION}/modules.builtin
-ln -sf ${NVIDIA}/lib/modules/${KERNEL_VERSION}/kernel /lib/modules/${KERNEL_VERSION}/.
-cd /root
-dnf -y group install "Development Tools"
-dnf -y install kernel-devel-${KERNEL_VERSION} kernel-headers-${KERNEL_VERSION} kmod binutils perl elfutils-libelf-devel
-git clone https://github.com/Mellanox/nv_peer_memory.git
-cd /root/nv_peer_memory
-sed -i 's/updates\/dkms/kernel\/drivers\/video/g' create_nv.symvers.sh
-./build_module.sh
-rpmbuild --rebuild /tmp/nvidia_peer_memory-*
-rpm  -ivh /root/rpmbuild/RPMS/x86_64/nvidia_peer_memory-*.rpm
-/etc/init.d/nv_peer_mem restart
-sleep infinity
\ No newline at end of file
+
+function set_driver_readiness() {
+    touch /.driver-ready
+}
+
+function unset_driver_readiness() {
+    rm -f /.driver-ready
+}
+
+function exit_on_error() {
+    $@
+    if [[ $? -ne 0 ]]; then
+        echo "ERROR: command execution failed: $1"
+        exit 1
+    fi
+}
+
+# has_files_matching() $1: dir path, $2: grep pattern
+function has_files_matching() {
+    local DIR_PATH=$1
+    local PATTERN=$2
+    ls $DIR_PATH 2> /dev/null | grep -E "${PATTERN}" > /dev/null
+    return $?
+}
+
+function install_prereq_runtime() {
+    echo "Checking for entitlement"
+    ls -l /etc/pki/entitlement-host
```

Comment:
This means we need to mount entitlements from host ?

Reply:
If entitlements are deployed then they're mounted in automatically on OpenShift, we don't need to do anything extra.